tests/test_detectAndMeasure.py

```diff
@@ -161,6 +164,35 @@ def test_detection_xy0(self):
         self.assertTrue(all(output.diaSources['id'] > 1000000000))
         self.assertImagesEqual(subtractedMeasuredExposure.image, difference.image)
 
+    def test_raise_bad_psf(self):
+        """Detection should raise if the PSF width is NaN
+        """
+        # Set up the simulated images
+        noiseLevel = 1.
+        staticSeed = 1
+        fluxLevel = 500
+        rng = np.random.RandomState(333)
+        kwargs = {"seed": staticSeed, "psfSize": 2.4, "fluxLevel": fluxLevel, "x0": 12345, "y0": 67890}
+        science, sources = makeTestImage(noiseLevel=noiseLevel, noiseSeed=6, **kwargs)
+        matchedTemplate, _ = makeTestImage(noiseLevel=noiseLevel/4, noiseSeed=7, **kwargs)
+        difference = science.clone()
+
+        psfShape = difference.getPsf().computeBBox(lsst.geom.Point2D(0, 0)).getDimensions()
+        psfcI = afwImage.ImageD(lsst.geom.Extent2I(psfShape[1], psfShape[0]))
+        psfNew = rng.randn(psfShape[1], psfShape[0])
+        psfcI.array = psfNew
+        psfcK = afwMath.FixedKernel(psfcI)
+        psf = measAlg.KernelPsf(psfcK)
+        difference.setPsf(psf)
+
+        # Configure the detection Task
+        detectionTask = self._setup_detection()
+
+        # Run detection and check the results
+        with self.assertRaises(UpstreamFailureNoWorkFound):
+            detectionTask.run(science, matchedTemplate, difference,
+                              idFactory=self.idGenerator.make_table_id_factory())
```

Comment:
No need to pass an idFactory unless you specifically care about the ids that are returned (which there won't be, since it raised).